src/contracts/strategies/EigenStrategy.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+// NOTE: Mainnet uses the OpenZeppelin v4.9.0 contracts, but this imports the 4.7.1 version. This will be changed after an upgrade.
```

Comment:
so is the plan to upgrade all contracts to 4.9.0 ?
it's definitely ugly to have both versions present at once

Reply:
We'd need to figure out how to deal with the EIP712 storage change

Reply:
i think the issue was primarily when doing a downgrade, upgrades are designed to be possible. but yes, good call - will require some review